plugin/web/extensions/sidebar/modals/datacatalog/components/content/DatasetsPage/DatasetTree/File.tsx

```diff
@@ -0,0 +1,103 @@
+import { CatalogItem } from "@web/extensions/sidebar/core/processCatalog";
+import { Button, Icon } from "@web/sharedComponents";
+import { styled } from "@web/theme";
+import { useCallback, useMemo } from "react";
+
+export type Props = {
+  item: CatalogItem;
+  addedDatasetIds?: string[];
+  nestLevel: number;
+  selected: boolean;
+  onDatasetAdd: (dataset: CatalogItem) => void;
+  onOpenDetails?: (item?: CatalogItem) => void;
+  onSelect?: (id: string) => void;
+};
+
+const File: React.FC<Props> = ({
+  item,
+  addedDatasetIds,
+  nestLevel,
+  selected,
+  onDatasetAdd,
+  onOpenDetails,
+  onSelect,
+}) => {
+  const handleClick = useCallback(() => {
+    onDatasetAdd(item);
+  }, [item, onDatasetAdd]);
+
+  const handleOpenDetails = useCallback(() => {
+    if (item.type === "group") return;
+    onOpenDetails?.(item);
+    onSelect?.(item.id);
+  }, [item, onOpenDetails, onSelect]);
+
+  const addDisabled = useMemo(() => {
+    return !!addedDatasetIds?.find(id => item?.type === "item" && id === item.id);
+  }, [addedDatasetIds, item]);
```

Comment:
You can keep it even shorter by doing something like
```
const addDisabled = useMemo(() => !!addedDatasetIds?.find(id => item?.type === "item" && id === item.id), [addedDatasetIds, item]);
```

Reply:
Done!